pkg/kubelet/cm/devicemanager/plugin/v1beta1/handler.go

```diff
@@ -83,7 +81,9 @@ func (s *server) connectClient(name string, socketPath string) error {
 }
 
 func (s *server) disconnectClient(name string) error {
+	s.mutex.Lock()
 	c := s.clients[name]
+	s.mutex.Unlock()
```

Comment:
This will not work because we need the lock around the call to `if _, exists := s.clients[pluginName]; exists {` (which is why it was there in the first place. This just introduces a new race.

Reply:
We should do a deep copy here if so.

Reply:
That won't help because we need to do the disconnect within the same lock as checking if the resource exists in the clients map.

Think the best thing to do (to fix the original issue and leave this (non-buggy) code path alone) is to just put an explicit lock/unlock around the call to disconnect in the `Stop()` function, i.e.:
```
func (s *server) Stop() error {
    for _, r := range s.clientResources() {
        s.mutex.Lock()
        if err := s.disconnectClient(r); err != nil {
            klog.InfoS("Error disconnecting device plugin client", "resourceName", r, "err", err)
        }
        s.mutex.Unlock()
    }
    ...
}     
```